src/extension/KaotoOutputChannel.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as vscode from 'vscode';
+
+export class KaotoOutputChannel {
+    private static instance: vscode.OutputChannel;
+
+    // Get or create the Kaoto Output Channel instance
+    public static getInstance(): vscode.OutputChannel {
+        if (!KaotoOutputChannel.instance) {
+            KaotoOutputChannel.instance = vscode.window.createOutputChannel('Kaoto');
+        }
+        return KaotoOutputChannel.instance;
+    }
+
+    public static logInfo(message: string): void {
+        this.logMessage('INFO', message);
+    }
+
+    public static logWarning(message: string): void {
+        this.logMessage('WARNING', message);
+    }
+
+    public static logError(message: string, error?: any): void {
+        const errorMsg = error instanceof Error ? error.message : String(error);
+        this.logMessage('ERROR', `${message}\n${errorMsg}`);
+    }
+
+    // Log a formatted message with a timestamp
+    private static logMessage(level: string, message: string): void {
+        const timestamp = new Date().toUTCString();
+        this.getInstance().appendLine(`[${timestamp}] [${level}] ${message}`);
+    }
+
+    // Dispose the Output Channel (mainly when the extension is deactivated)
+    public static dispose(): void {
+        if (KaotoOutputChannel.instance) {
+            KaotoOutputChannel.instance.dispose();
+        }
+    }
```

Comment:
I think we need to also set back `KaotoOutputChannel.instance` to `undefined`. Otherwise when extensions are restarted, it will fail next time it tries to log something

Reply:
I was not able to simulate using `Reload Window` nor `Extensions: Refresh` nor disable/enable Kaoto extension

Reply:
I needed to build the vsix and install it to go through activate/deactivate of the extension. (in dev mode the disable/enable is using the installed extension, not the one under development)
But just tried and it is working anyway. i do not understand why the `KaotoOutputChannel.instance` when disposed is not throwing an error the next time it is called